deluge/ui/console/main.py

```diff
@@ -238,15 +239,28 @@ def __init__(self, options=None, cmds=None):
         # Set the interactive flag to indicate where we should print the output
         self.interactive = True
         self._commands = cmds
-        if options.parsed_cmds:
+        self.coreconfig = CoreConfig()
+
+    def start_ui(self):
+        """Start the console UI.
+
+        Note: When running console UI reactor.run() will be called which
+              effectively blocks this function making the return value
+              insignificant. However, when running unit tests, the reacor is
+              replaced by a mock object, leaving the return deferred object
+              necessary for the tests to run properly.
+
+        Returns: Deferred that fires when all commands are executed.
+        """
+        deferred = None
+        if self.options.parsed_cmds:
             self.interactive = False
-            if not cmds:
+            if not self._commands:
```

Comment:
Think that this can be refactored to bring the reactor.run and return into this statement.